lib/core_modules/module/aggregation.js

```diff
@@ -484,61 +484,57 @@ function supportAggregate(Meanio) {
       options.separator = '?v=';
     }
 
-    var self = this;
-    this.settings(function(err, settings) {
-      if (settings &&
-        settings.settings &&
-        settings.settings.languages &&
-        settings.settings.currentLanguage) {
-
-        var languages = settings.settings.languages;
-        var currentLanguage = settings.settings.currentLanguage;
-
-        currentLanguage = _(languages).find(function(l) {
-          return l.locale === currentLanguage;
-        });
-
-        if (currentLanguage) {
-          var direction = currentLanguage.direction;
-          if (direction) {
-            options.direction = direction;
-          }
+    if (Meanio.Singleton.config.clean.public && 
+      Meanio.Singleton.config.clean.public.languages &&
+      Meanio.Singleton.config.clean.public.currentLanguage) {
+
+      var languages = Meanio.Singleton.config.clean.public.languages;
+      var currentLanguage = Meanio.Singleton.config.clean.public.currentLanguage;
+
+      currentLanguage = _(languages).find(function(l) {
+        return l.locale === currentLanguage;
+      });
+
+      if (currentLanguage) {
+        var direction = currentLanguage.direction;
+        if (direction) {
+          options.direction = direction;
         }
       }
+    }
 
-      if (options.inline){
-        switch(type){
-          case 'css':
-            new InlineCSSAsset(asset,options);
-            return;
-          case 'js':
-            new InlineJSAsset(asset,options);
-            return;
-        }
-        return;
+    if (options.inline){
+      switch(type){
+        case 'css':
+          new InlineCSSAsset(asset,options);
+          return;
+        case 'js':
+          new InlineJSAsset(asset,options);
+          return;
       }
-      else if (options.url){
-        switch(type){
-          case 'css':
-            new RemoteCSSAsset(asset,options);
-            return;
-          case 'js':
-            new RemoteJSAsset(asset,options);
-            return;
-        }
-        return;
-      }else{
-        var fpath = path.join(self.loadedmodule.path('public'),'assets',type,asset);
-        switch(type){
-          case 'css':
-            new CSSFileAsset(fpath,path.join(self.name,'assets',type,asset),options);
-            return;
-          case 'js':
-            new JSFileAsset(fpath,path.join(self.name,'assets',type,asset),options);
-            return;
-        }
+      return;
+    }
+    else if (options.url){
+      switch(type){
+        case 'css':
+          new RemoteCSSAsset(asset,options);
+          return;
+        case 'js':
+          new RemoteJSAsset(asset,options);
+          return;
       }
-    });
+      return;
+    }else{
+      var fpath = path.join(this.loadedmodule.path('public'),'assets',type,asset);
+      switch(type){
+        case 'css':
+          new CSSFileAsset(fpath,path.join(this.name,'assets',type,asset),options);
```

Comment:
Line is too long.<br>Do not use 'new' for side effects.<br>Too many errors. (88% scanned).